### PR TITLE
Added L2 norm regularization term to the loss to prevent gradient exp…

### PR DESCRIPTION
**Pull request by r-zemblys**
Fixes [#65](https://github.com/ibab/tensorflow-wavenet/issues/65) issue with gradient explosion

### CODE REVIEW

File: wavenet.py [hunk 1]
```diff
@@ -235,7 +235,16 @@ def loss(self, input_batch, name='wavenet'):
...
+                                    for v in tf.trainable_variables()])
+                
+                # Add the regularization term to the loss
+                total_loss=reduced_loss + C * l2_loss
 
                 tf.scalar_summary('loss', reduced_loss)
+                tf.scalar_summary('l2_loss', l2_loss)
+                tf.scalar_summary('total_loss', total_loss)
 
         return reduced_loss
```
Comment (jyegerlehner) [inline wavenet.py]: Wouldn't we want to be returning total_loss here?

File: wavenet.py [hunk 2]
```diff
@@ -211,7 +211,7 @@ def predict_proba(self, waveform, name='wavenet'):
             return tf.reshape(last, [-1])
 
 
-    def loss(self, input_batch, name='wavenet'):
+    def loss(self, input_batch, C=5e-4, name='wavenet'):
```
Comment (jyegerlehner) [inline wavenet.py]: C looks like something we would want to configure from command line or wavenet_params.json.

Comment (jyegerlehner) [inline wavenet.py]: ...and I might rather have that default to zero.

File: wavenet.py (hunk 1 above)
Comment (r-zemblys) [inline wavenet.py]: Right, sorry about that

File: wavenet.py (hunk 2 above)
Comment (r-zemblys) [inline wavenet.py]: I would add it as a command line parameter. But defaulting it to zero is a good idea

**Comment by r-zemblys** [comment]
I'm closing this PR as I messed up with my branch
